test/test_torch.py

```diff
@@ -12484,112 +12484,140 @@ def unsqueeze_op_clone(x, y):
+                                    count += 1
+                            self.assertEqual(j, count)
+
+                    # Tests per-element unique on a higher rank tensor.
+                    y = x.view(2, 2, 2)
+                    y_unique, y_inverse, y_counts = torch.unique(
+                        y, sorted=True, return_inverse=True, return_counts=True)
+                    self.assertEqual(expected_unique, y_unique)
+                    self.assertEqual(expected_inverse.view(y.size()), y_inverse)
+                    self.assertEqual(expected_counts, y_counts)
+
+        # test scalar
+        x = torch.tensor(0, dtype=dtype, device=device)
+        for sorted_ in [True, False]:
+            unique, inverse, counts = torch.unique(x, sorted=sorted_, return_inverse=True, return_counts=True)
+            expected_unique = torch.tensor([0], dtype=dtype, device=device)
+            expected_inverse = torch.tensor(0, device=device)
+            expected_counts = torch.tensor([1], device=device)
+
+        # test zero sized tensor
+        x = torch.zeros((0, 0, 3), dtype=dtype, device=device)
+        for sorted_ in [True, False]:
+            unique, inverse, counts = torch.unique(x, sorted=sorted_, return_inverse=True, return_counts=True)
+            expected_unique = torch.tensor([], dtype=dtype, device=device)
+            expected_inverse = torch.empty((0, 0, 3), dtype=torch.long, device=device)
+            expected_counts = torch.tensor([], dtype=torch.long, device=device)
+
+    @dtypes(*set(torch.testing.get_all_dtypes()) - {torch.bfloat16})
+    def test_unique_consecutive(self, device, dtype):
+        if dtype is torch.half and self.device_type == 'cpu':
+            return  # CPU does not have half support
+
+        def ensure_tuple(x):
+            if torch.is_tensor(x):
+                return (x,)
+            return x
+
+        if dtype is torch.bool:
+            bool_unique, bool_inverse, bool_counts = torch.unique_consecutive(
+                torch.tensor([True, False, False, False, True, True, False], dtype=torch.bool, device=device),
+                return_inverse=True,
+                return_counts=True
+            )
+            self.assertEqual(torch.tensor([True, False, True, False], dtype=torch.bool, device=device), bool_unique)
+            self.assertEqual(torch.tensor([0, 1, 1, 1, 2, 2, 3], dtype=torch.long, device=device), bool_inverse)
+            self.assertEqual(torch.tensor([1, 3, 2, 1], dtype=torch.long, device=device), bool_counts)
+        else:
+            # test consecutive version
+            x = torch.tensor([1, 2, 2, 2, 5, 5, 2, 2, 3], dtype=dtype, device=device)
+            expected_unique = torch.tensor([1, 2, 5, 2, 3], dtype=dtype, device=device)
+            expected_inverse = torch.tensor([0, 1, 1, 1, 2, 2, 3, 3, 4], device=device)
+            expected_counts = torch.tensor([1, 3, 2, 2, 1], device=device)
+
+            for return_inverse in [True, False]:
```

Comment:
There's a lot of copy-paste between unique_consecutive and unique tests, I feel you can factor it out into helper functions.